### ISSUE TITLE
customize: expose `offsetTernaryExpressions`

### ISSUE THREAD

**Issue by teidesu** [eslint-stylistic/eslint-stylistic#514]
### Clear and concise description of the problem

i want the `customize` function to allow customizing `offsetTernaryExpressions` option in `@stylistic/indent`

### Suggested solution

expose it in `StylisticCustomizeOptions` [here](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/packages/eslint-plugin/configs/customize.ts)

### Alternative

a possible alternative would be to fix `style/indent` to make `offsetTernaryExpressions` look good on indents other than 2 (i.e. indent the closing `}` using 2 spaces regardless of the global indent)

another possible alternative would be to change the default value for `offsetTernaryExpressions` in customize function to `indent === 2`

### Additional context

the main reason is that when the indent used is not default (e.g. 4), having `offsetTernaryExpressions` looks really bad.

indent = 4, offsetTernaryExpressions = true:

```ts
const a = Math.random() > 0.5
    ? {
            foo: 123,
        }
    : {
            bar: 456,
        }
```

indent = 4, offsetTernaryExpressions = false:

```ts
const a = Math.random() > 0.5
    ? {
        foo: 123,
    }
    : {
        bar: 456,
    }
```

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [X] If this feature request is accepted, I am willing to submit a PR to fix this issue
